### PR TITLE
document ini resource

### LINKED ISSUE #848: Document ini resource
As a developer I would like to verify ini files within inspec:
- if given sections exist
- if given keys+values exist within sections

This should be as new resource, similar to parse_config.

### PR DESCRIPTION
**Pull request by vjeffrey**
let me know if this is ok?  i just kinda best-guessed it.  @chris-rock 

fixes #848

### CODE REVIEW

File: docs/resources.rst [hunk 1]
```diff
@@ -1999,6 +2000,48 @@ then the same test will return ``false`` for ``ftp`` and the entire test will fa
...
+-----------------------------------------------------
+The following examples show how to use this InSpec audit resource.
+
+For example:
+
+.. code-block:: ruby
+
+   describe ini do
+     its('port') { should eq 143 }
+     its('server') { should eq 192.0.2.62 }
```
Comment (ksubrama) [inline docs/resources.rst]: I'm curious...  does this work?  I'd think it'd look like '192.0.2.63'

File: docs/resources.rst [hunk 2]
```diff
@@ -1999,6 +2000,48 @@ then the same test will return ``false`` for ``ftp`` and the entire test will fa
...
+
+**Stability: Stable**
+
+Syntax
+-----------------------------------------------------
+An ``ini`` |inspec resource| block declares the content of the ``ini`` file:
+
+.. code-block:: ruby
+
+   describe ini do
```
Comment (ksubrama) [inline docs/resources.rst]: describe ini("path/to/ini_file.ini")

File: docs/resources.rst [hunk 3]
```diff
@@ -1999,6 +2000,48 @@ then the same test will return ``false`` for ``ftp`` and the entire test will fa
...
+
+Examples
+-----------------------------------------------------
+The following examples show how to use this InSpec audit resource.
+
+For example:
+
+.. code-block:: ruby
+
+   describe ini do
```
Comment (ksubrama) [inline docs/resources.rst]: ditto

File: docs/resources.rst [hunk 4]
```diff
@@ -1999,6 +2000,48 @@ then the same test will return ``false`` for ``ftp`` and the entire test will fa
...
+* ``'auth_protocol'`` is a key in the ``ini`` file
+* ``('https')`` is the expected value associated with the above key in the ``ini`` file
+
+Matchers
+-----------------------------------------------------
+This |inspec resource| matches any content in the ``ini`` file:
+
+.. code-block:: ruby
+
+   its('port') { should eq 143 }
```
Comment (ksubrama) [inline docs/resources.rst]: I think this should be '143'

**Comment by ksubrama** [comment]
@vjeffrey If you want to test this out try this:

```
$ cat ../foo.ini
ipaddress=192.168.1.1
foo= 45
$ bundle exec inspec shell -c "describe ini('../foo.ini') do its('ipaddress') {should eq '192.168.1.1'} end"
  ✔  INI ../foo.ini ipaddress should eq "192.168.1.1"

Summary: 1 successful, 0 failures, 0 skipped
$ bundle exec inspec shell -c "describe ini('../foo.ini') do its('foo') {should eq 45} end"
  ✖  INI ../foo.ini foo should eq 45 (
     expected: 45
          got: "45"

     (compared using ==)
     )

Summary: 0 successful, 1 failures, 0 skipped
```

**Comment by vjeffrey** [comment]
oh thanks! i'll fix that all up later this afternoon :)

**Comment by chris-rock** [comment]
Thanks @vjeffrey for the improvement and @ksubrama for the great review

**Pull request merged**